packages/editor/src/components/post-title/index.native.js

```diff
@@ -22,6 +22,13 @@ import { pasteHandler } from '@wordpress/blocks';
 import styles from './style.scss';
 
 class PostTitle extends Component {
+	componentDidUpdate( prevProps ) {
+		// Unselect if any other block is selected
+		if ( this.props.isSelected && ! prevProps.isAnyBlockSelected && this.props.isAnyBlockSelected ) {
+			this.props.onUnselect();
```

Comment:
It doesn't feel right to me that we have a _component_ that is essentially checking the state of the store (`isPostTitleSelected` and `isAnyBlockSelected`) to determine whether/how to make a further change to the store (updating `isPostTitleSelected`). To me that sounds like the kind of job that a reducer is for—take the previous store state (`isPostTitleSelected`) and an action (a block selection action) and update the store.

What do you think?

Reply:
Problem here is that `isAnyBlockSelected` comes from the `core/block-editor` store and we're updating the `core/editor` store here so a reducer would not work. I don't believe we should use controls or resolvers either as that would introduce a dependency between the stores.

I guess the actual fix would be to make sure that whenever a block is focused, a blur event is triggered on the post title, listen to that event and update the stateS accordingly

Reply:
I keep forgetting about the multiple stores. I'm fine with this approach in light of that.